features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5_SDK13/source/btle/btle_security_pm.cpp

```diff
@@ -434,4 +434,59 @@ btle_generateResolvableAddress(const ble_gap_irk_t &irk, ble_gap_addr_t &address
     /* Calculate the hash and store it in the top half of the address */
     ah(irk.irk, &address.addr[BLE_GAP_ADDR_LEN - 3], address.addr);
 }
+
+
+#if (NRF_SD_BLE_API_VERSION >= 3)
+ble_error_t btle_getAddressesFromBondTable(Gap::Whitelist_t &addrList)
+{
+    pm_peer_id_t           peer_id;
+    ret_code_t             ret;
+    pm_peer_data_bonding_t bond_data;
+
+    addrList.size = 0;
+    peer_id       = pm_next_peer_id_get(PM_PEER_ID_INVALID);
+
+    /**
+     * Fill addresses list:
+     * Copy addresses from bond table, or
+     * for every private resolvable address in the bond table check if
+     * there is maching address in th provided whitelist and copy resolvable address.
+     */
+    while (peer_id != PM_PEER_ID_INVALID) {
+        memset(&bond_data, 0x00, sizeof(bond_data));
+
+        // Read peer data from flash.
+        ret = pm_peer_data_bonding_load(peer_id, &bond_data);
+
+        if ((ret == NRF_ERROR_NOT_FOUND) || (ret == NRF_ERROR_INVALID_PARAM)) {
+            // Peer data could not be found in flash or peer ID is not valid.
+            return BLE_ERROR_UNSPECIFIED;
+        }
+
+        if (addrList.capacity <= addrList.size) {
+            /* Ran out of space in the output Gap::Whitelist_t */
+            return BLE_ERROR_NONE;
+        }
```

Comment:
When there is no more space in the output list, there is no point in reading the bonding data. Hence, I'd move this checking to `while` condition, and change it to `addrList.size < addrList.capacity` (IMHO this would better express what is actually checked).